homework-submission/myapp/src/stores/rashadStore.js

```diff
@@ -0,0 +1,93 @@
+import { configure, action, observable, runInAction, computed } from "mobx";
+import Todos from "../components/todos.json";
+
+configure({ enforceActions: true });
+
+const url = "https://hyf-react-api.herokuapp.com";
+
+class RashadStore {
+  @observable todosList = [];
+
+  fetchAll = () => {
+    fetch(`${url}/todos`)
+      .then(response => response.json())
+      .then(todosList => {
+        console.log(todosList);
+        runInAction(() => {
+          this.todosList = Todos;
+        });
+      })
+      .catch(err => console.log(err));
+  };
+
+  @action
+  checker = item => {
+    item.done = !item.done;
```

Comment:
Following the code it's nice passing till here 😄 But this is a ~> _read only props_
means it will modify the reference of the props (_you or MobX passed already_) while it's read only.
two ways to solve the problem you have:

1. according to this action with a normal mutating for the `observable` __Without fetch__ you have to modify the `observable` it self.

2. with fetch !! __What If The Fetch Got An Error Back?__ so istead of mutating it Accidentally, you will fake it and then nothing will happen to the data we want 💔 Solution ? ~> Look the next comment